Routers/Routers/Routers/Routers.cs

```diff
@@ -0,0 +1,290 @@
+﻿namespace Routers;
+
+using System;
+using System.Collections.Generic;
+
+/// <summary>
+/// Class representing a graph
+/// </summary>
+public class Graph : IGraph
+{
+    /// <summary>
+    /// Class for storing graph vertices
+    /// </summary>
+    private class Node
+    {
+        // Useless dictionary needed to output a table to a file by condition
+        public Dictionary<int, int> newDictionary = new();
+    }
+
+    public Graph(){ }
+
+    public Graph (int x)
+    {
+        AddNode(x);
+    }
+
+    /// List for storing vertices
+    private readonly List<Node> nodes = new();
+
+    /// List for storing vertex values
+    /// nodes[0] corresponds to nodeValue[0]
+    private readonly List<int> nodeValue = new();
+
+    /// Dictionary of Adjacency
+    /// The key is a pair of vertices, the value is the length of the edge between them
+    private Dictionary<(int, int), int> adjacencyDictionary = new();
+
+    /// <summary>
+    /// Function for adding a node
+    /// </summary>
+    /// <param name="index">Node value</param>
+    /// <returns>Was there a value in the graph</returns>
+    public bool AddNode(int index)
+    {
+        if (!nodeValue.Contains(index))
+        {
+            nodes.Add(new());
+            nodeValue.Add(index);
+            return true;
+        }
+
+        return false;
+    }
+
+    /// <summary>
+    /// Function for creating an edge between vertices
+    /// </summary>
+    /// <param name="valueFirstNode">Value of the first node</param>
+    /// <param name="valueSecondNode">Value of the second node</param>
+    /// <param name="length">Edge length</param>
+    /// <returns>Was there an edge in the graph</returns>
+    public bool SetLength(int valueFirstNode, int valueSecondNode, int length)
+    {
+        if (!nodeValue.Contains(valueFirstNode) || !nodeValue.Contains(valueSecondNode) || valueSecondNode == valueFirstNode)
+        {
+            return false;
+        }
+
+        if (adjacencyDictionary.ContainsKey((valueFirstNode, valueSecondNode)))
+        {
+            adjacencyDictionary.Remove((valueFirstNode, valueSecondNode));
+        }
+
+        adjacencyDictionary.Add((valueFirstNode, valueSecondNode), length);
```

Comment:
```suggestion
        adjacencyDictionary[(valueFirstNode, valueSecondNode)] = length;
```
Индексер не кидает исключение, если элемент уже есть, а это как раз то, что Вам и надо